### PR TITLE
42 add H block with tests

### PR DESCRIPTION
**Pull request by Slawek-Sandbox**
Implemented H block (including tests)

### CODE REVIEW

File: src/test/java/com/epam/prejap/tetris/block/HBlockTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package com.epam.prejap.tetris.block;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+@Test(groups = {"blockShapes"})
```
Comment (TJB4epam) [inline src/test/java/com/epam/prejap/tetris/block/HBlockTest.java:8]: I'm a bit sick of those tests. Why don't we have a proper solution for once, with data providers, factories and one that ELIMINATES all the WhateverBlockTest classes and makes a parametrized test for them all?

**Review by TJB4epam — CHANGES_REQUESTED** [review]
1. Please update the sources and resolve conflicts.
2. Where's the POM version bump?
3. See review remarks.

**Comment by TJB4epam** [comment]
Conflicts reminder, please resolve.

File: src/test/java/com/epam/prejap/tetris/block/HBlockTest.java (hunk 1 above)
Comment (Slawek-Sandbox) [reply src/test/java/com/epam/prejap/tetris/block/HBlockTest.java]: I see @asilenko have already fixed that. I'm sorry - didn't have time to do it myself quicker.

**Comment by Slawek-Sandbox** [comment]
I addressed all remarks
Ad 1 Conflicts resolved
Ad 2 POM version bump in
Ad 3 Didn't implement TestNG factory - I see Anna did, but I have refactored BlocksShapesTest class. I hope it counts ;)

**Review by TJB4epam — CHANGES_REQUESTED** [review]
TY for the version.

Block, instead of String name and actual rows/cols is good for readability.
String.format - excellent choice!

Please, move the factory/dp combo that powers this class up - these are like a c-tor, so should be right after fields.

Actual tests should happen lower.

Overall - a good improvement on Anna's solution, that focuses on readability. I'd like to merge it fast.

**Comment by Slawek-Sandbox** [comment]
TY for the review.

I moved factory/dp combo up, so test are at the bottom right now. I hope that's what you requested (I was not sure about private helper methods).

**Comment by Slawek-Sandbox** [comment]
Summary:
- moved factory/dp combo up
- resolved conflicts (which appeared after [5-i-implement-i-block](https://github.com/EPAM-Online-Courses/tetris/pull/73) was merged)
- bumped POM version again

PR should be ready to merge.

**Review by TJB4epam — APPROVED** [review]
Thanks!

**Pull request merged**